Documentation/teaching/labs/kernel_api.rst

```diff
@@ -584,99 +584,132 @@ Using |LXR|_ find the definitions of the following symbols in the Linux kernel:
    * :c:type:`struct list_head`
    * :c:type:`INIT_LIST_HEAD`
    * :c:func:`list_add`
-   * :c:func:`list_for_each`
-   * :c:func:`list_entry`
-   * :c:func:`container_of`
-   * :c:func:`offsetof`
+   * :c:macro:`list_for_each`
+   * :c:macro:`list_entry`
+   * :c:macro:`container_of`
+   * :c:macro:`offsetof`
 
 1. Memory allocation in Linux kernel
 ------------------------------------
 
 Generate the skeleton for the task named **1-mem** and browse the
-contents of the ``mem.c`` file.  Observe the use of kmalloc call for
-memory allocation.
+contents of the :file:`mem.c` file.  Observe the use of :c:func:`kmalloc`
+call for memory allocation.
 
-   1. Compile the source code and load the ``mem.ko`` module using ``insmod``.
-   2. View the kernel messages using the ``dmesg`` command.
-   3. Unload the kernel module using the ``rmmod mem`` command.
+   1. Compile the source code and load the :file:`mem.ko` module using
+      :command:`insmod`.
+   2. View the kernel messages using the :command:`dmesg` command.
+   3. Unload the kernel module using the :command:`rmmod mem` command.
 
 .. note:: Review the `Memory Allocation`_ section in the lab.
 
 2. Sleeping in atomic context
 -----------------------------
 
 Generate the skeleton for the task named **2-sched-spin** and browse
-the contents of ``sched-spin.c`` file.
+the contents of the :file:`sched-spin.c` file.
 
-   1. Compile the source code and load the module.
+   1. Compile the source code and load the module, according the above info:
+      (:command:`make info` and :command:`make copy`)
    2. Notice that it is waiting for 5 seconds until the insertion
       order is complete.
-   3. Unload the kernel mode.
-   4. Look for the lines marked with TODO0 to create an atomic
+   3. Unload the kernel module.
+   4. Look for the lines marked with: ``TODO 0`` to create an atomic
       section. Re-compile the source code and reload the module into
       the kernel.
 
 You should now get an error. Look at the stack trace. What is the
 cause of the error?
 
-.. hint:: In the error message, follow the line containing the BUG for
-	  a description of the error. You are not allowed to sleep in
+.. hint:: In the error message, follow the line containing the :c:macro:`BUG`
+          for a description of the error. You are not allowed to sleep in
 	  atomic context.  The atomic context is given by a section
 	  between a lock operation and an unlock on a spinlock.
 
-.. note:: The schedule_timeout function, corroborated with the
-	  set_current_state macro, forces the current process to wait
-	  S seconds.
+.. note:: The
+          `schedule_timeout <https://elixir.bootlin.com/linux/v4.15.7/source/kernel/time/timer.c#L1725>`_
```

Comment:
Please don't use links to LXR, we should use the :c:func: notation since we want to have this integrated with the current kernel doc system.